### PR TITLE
Add eth_signTypedData method.

### LINKED ISSUE #83: Add eth_signTypedData_v4 Support
Some wallets, such as Uniswap, don't support `personal_sign`. Would be nice to use the latest `eth_signTypedData_v4` to get pretty messages.

### PR DESCRIPTION
**Pull request by vuongxuantuyen**
# Description
Add ethnic_signTypedData method.

Resolves #83 (issue)

## Due Dilligence

* [ ] Breaking change
* [x] Requires a documentation update

### CODE REVIEW

File: Sources/Web3Modal/Core/W3MJSONRPC.swift [hunk 1]
```diff
@@ -7,6 +7,12 @@ public enum W3MJSONRPC: Codable {
         address: String,
         message: String
     )
+
+    case eth_signTypedData(
```
Comment (mooshee) [inline Sources/Web3Modal/Core/W3MJSONRPC.swift:11]: shouldn't this be `eth_signTypedData_v4`?

File: Sources/Web3Modal/Core/Web3ModalClient.swift [hunk 2]
```diff
@@ -276,7 +281,12 @@ public class Web3ModalClient {
             break
         }
     }
-    
+
+    public func removeSessionAndAccount() {
+        store.session = nil
+        store.account = nil
+    }
```
Comment (mooshee) [inline Sources/Web3Modal/Core/Web3ModalClient.swift:288]: Why is this here?

**Comment by mooshee** [comment]
While this PR adds the necessary methods, it doesn't make use of it. The `WalletDetailViewModel.signSIWE()` needs to check if the session supports `eth_signTypedData_v4` and then make the request instead of using `personal_sign`. Furthermore, the signature validation is more complex. A new EIP712Verifier needs to be added to MessageVerifier.

File: Sources/Web3Modal/Core/Web3ModalClient.swift (hunk 2 above)
Comment (vuongxuantuyen) [reply Sources/Web3Modal/Core/Web3ModalClient.swift:288]: @mooshee My mistake, just a test function. This should be removed.

**Review by GloWE3 — COMMENTED** [review]
image
